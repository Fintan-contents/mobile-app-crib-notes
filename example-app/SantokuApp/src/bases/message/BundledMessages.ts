export const bundledMessages = {
  ホーム: 'ホーム',
  ユーザー: 'ユーザー',
  質問詳細: '質問詳細',
  プロフィール: 'プロフィール',
  利用規約: '利用規約',
  リロード: 'リロード',
  同意: '同意',
  閉じる: '閉じる',
  OK: 'OK',
  アプリアップデート: 'アプリアップデート',
  システムエラー: 'システムエラー',
  ログイン: 'ログイン',
  新規登録: '新規登録',
  アカウントID: 'アカウントID',
  パスワード: 'パスワード',
  プロフィール登録: 'プロフィール登録',
  ニックネーム: 'ニックネーム',
  登録: '登録',
  ログイン失敗: 'ログイン失敗',
  ログアウト: 'ログアウト',
  消去: '消去',
  キャンセル: 'キャンセル',
  完了: '完了',
  年: '年',
  月: '月',
  ページ番号: 'ページ番号',
  タイトル: 'タイトル',
  コメント: 'コメント',
  コメントを入力: 'コメントを入力',
  説明: '説明',
  質問: '質問',
  回答: '回答',
  最新順: '最新順',
  いいね順: 'いいね順',
  更新日順: '更新日順',
  件: '件',
  解決済み: '解決済み',
  日前: '日前',
  時間前: '時間前',
  質問は削除されました: '質問は削除されました',
  コメントを追加する: 'コメントを追加する',
  投稿: '投稿',
  質問を投稿: '質問を投稿',
  イベントを告知: 'イベントを告知',
  募集中のイベント: '募集中のイベント',
  もっと見る: 'もっと見る',
  クリア: 'クリア',
  決定: '決定',
  初歩的な質問内容の場合はチェックしてください: '初歩的な質問内容の場合はチェックしてください',
  'TISインテックグループで共同利用されるため、社外秘情報（画像含む）を投稿しないでください':
    'TISインテックグループで共同利用されるため、社外秘情報（画像含む）を投稿しないでください',
  'アカウントIDまたはパスワードに\n間違いがあります。': 'アカウントIDまたはパスワードに\n間違いがあります。',
  '現在のバージョン({0})ではご利用いただけません。ストアより最新版アプリへのアップデートをお願いします。':
    '現在のバージョン({0})ではご利用いただけません。ストアより最新版アプリへのアップデートをお願いします。',
  '通信中にエラーが発生しました。': '通信中にエラーが発生しました。',
  'しばらく経ってからもう1度お試しください。': 'しばらく経ってからもう1度お試しください。',
  'validation.mixed.required': '${path}を入力してください。',
  'validation.string.max': '${max}文字以下で入力してください。',
  'validation.string.loginPassword': '${path}が正しくありません。',
  'validation.number.max': '${path}は${max}以下で入力してください。',
  'app.webview.onError': '通信エラーが発生し、コンテンツの取得に失敗しました。ネットワーク環境を確認してください。',
  'app.account.signupError': 'サインアップ後のログインに失敗しました。cause=[{0}]',
  'app.account.サインアップエラータイトル': 'サインアップに失敗しました',
  'app.account.サインアップエラー本文': '時間をおいてから再度お試しください。',
  'app.account.ログアウトエラータイトル': 'ログアウトに失敗しました',
  'app.account.ログアウトエラー本文': '時間をおいてから再度お試しください。',
  'app.terms.有効な利用規約の取得エラー': '利用規約の取得に失敗しました。画面をリロードして再度お試しください。',
  'app.push.notification.deleteFcmTokenError': 'FCM登録トークンの削除に失敗しました。cause=[{0}]',
  'app.push.notification.プッシュ通知の設定エラー':
    'プッシュ通知の設定に失敗しました。\n現在、プッシュ通知を受信できません。',
  'fw.error.通信エラー': 'ページの表示中にエラーが発生しました。ネットワーク環境を確認してください。',
  'fw.error.再ログインタイトル': '再ログインが必要です',
  'fw.error.再ログイン本文': 'セッションの有効期限が切れました。再度ログインしてください。',
  'fw.error.アプリバージョンエラータイトル': 'アプリの更新が必要です',
  'fw.error.アプリバージョンエラー本文':
    'アプリのバージョンが古いためこの機能を利用できません。ストアからアップデートを実施してください。',
  'fw.error.利用規約未同意タイトル': '新しい利用規約への同意',
  'fw.error.利用規約未同意本文': 'この機能を利用するためには最新の利用規約への同意が必要です。',
  'fw.error.リクエスト過多': 'ただいまリクエストが集中しており混雑しております。時間をおいてから再度お試しください。',
  'fw.error.リクエストタイムアウト': 'サーバへの接続がタイムアウトしました。時間をおいてから再度お試しください。',
  'fw.error.システムメンテナンス':
    'ただいまシステムメンテナンスを実施しております。時間をおいてから再度お試しください。',
  'fw.error.予期せぬ通信エラー':
    '予期せぬ通信エラーが発生しました。時間をおいてから再度お試しいただき、解決しない場合はお問い合わせください。',
} as const;

export type BundledMessagesType = typeof bundledMessages;
export type MessageKey = keyof typeof bundledMessages;
