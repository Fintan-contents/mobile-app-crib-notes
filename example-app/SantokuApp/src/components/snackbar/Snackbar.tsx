import React, {useCallback, useRef, useState} from 'react';
import {Animated, StyleProp, StyleSheet, Text, TextStyle, TouchableOpacity, View, ViewStyle} from 'react-native';

import {FullWindowOverlay} from './FullWindowOverlay';

import CompositeAnimation = Animated.CompositeAnimation;

export type SnackbarShowProps = {
  message: string;
  messageTextStyle?: StyleProp<TextStyle>;
  top?: number;
  bottom?: number;
  right?: number;
  left?: number;
  style?: StyleProp<ViewStyle>;
  animatedContainerStyle?: StyleProp<ViewStyle>;
  actionText?: string;
  actionHandler?: () => void;
  actionTextStyle?: StyleProp<TextStyle>;
  autoHideDuration?: number;
  fadeInDuration?: number;
  fadeOutDuration?: number;
  forceFadeOutDuration?: number;
};

export type SnackbarHideProps = {
  hide?: true;
  hideFadeOutDuration?: number;
};

export type SnackbarProps = SnackbarShowProps & SnackbarHideProps;

export const Snackbar: React.FC<SnackbarProps> = (props) => {
  const fadeAnim = useRef(new Animated.Value(0)).current;
  const fadeInAnimationRef = useRef<CompositeAnimation>();
  const fadeOutAnimationRef = useRef<CompositeAnimation>();
  const barrierFadeOutAnimationRef = useRef<CompositeAnimation>();
  const [visibleSnackbarProps, setVisibleSnackbarProps] = useState<SnackbarShowProps>();

  const animationStart = useCallback(
    (
      animationRef: React.MutableRefObject<CompositeAnimation | undefined>,
      config: Animated.TimingAnimationConfig,
      callback?: Animated.EndCallback,
    ) => {
      const animation = Animated.timing(fadeAnim, config);
      animationRef.current = animation;
      animation.start((result) => {
        animationRef.current = undefined;
        if (callback) {
          callback(result);
        }
      });
    },
    [fadeAnim],
  );

  const show = useCallback(() => {
    setVisibleSnackbarProps(props);

    const fadeInAnimationConfig = {
      toValue: 1,
      duration: props.fadeInDuration,
      useNativeDriver: true,
    };
    animationStart(fadeInAnimationRef, fadeInAnimationConfig, ({finished}) => {
      if (finished) {
        const fadeOutAnimationConfig = {
          toValue: 0,
          delay: props.autoHideDuration,
          duration: props.fadeOutDuration,
          useNativeDriver: true,
        };
        animationStart(fadeOutAnimationRef, fadeOutAnimationConfig, () => {
          if (!barrierFadeOutAnimationRef) {
            setVisibleSnackbarProps(undefined);
          }
        });
      } else {
        if (!barrierFadeOutAnimationRef) {
          setVisibleSnackbarProps(undefined);
        }
      }
    });
  }, [props, animationStart]);

  const forceFadeout = useCallback(
    (fadeOutDuration?: number, callback?: () => void) => {
      const barrierFadeOutAnimationConfig = {
        toValue: 0,
        duration: fadeOutDuration,
        useNativeDriver: true,
      };

      fadeInAnimationRef.current?.stop();
      fadeOutAnimationRef.current?.stop();

      animationStart(barrierFadeOutAnimationRef, barrierFadeOutAnimationConfig, () => {
        setVisibleSnackbarProps(undefined);
        callback?.();
      });
    },
    [animationStart],
  );

  React.useEffect(() => {
    if (props.hide) {
      forceFadeout(props.hideFadeOutDuration ?? props.forceFadeOutDuration);
      return;
    }
    if (!props.message) {
      return;
    }
    if (barrierFadeOutAnimationRef.current) {
      return;
    }
    if (fadeInAnimationRef.current || fadeOutAnimationRef.current) {
      forceFadeout(props.forceFadeOutDuration, show);
      return;
    }
    show();
  }, [props, show, forceFadeout]);

  const snackbarStyle = StyleSheet.flatten([styles.snackbar, props.style]);

  const animatedViewStyle = StyleSheet.flatten<ViewStyle>([styles.animatedContainer, props.animatedContainerStyle]);
  if (animatedViewStyle?.top === undefined && animatedViewStyle.bottom === undefined) {
    animatedViewStyle.bottom = 20;
  }

  return (
    <>
      {props.children}
      {visibleSnackbarProps && (
        <FullWindowOverlay>
          <Animated.View style={StyleSheet.flatten([{opacity: fadeAnim}, animatedViewStyle])}>
            <View style={snackbarStyle}>
              <View style={styles.messageContainer}>
                <Text style={styles.messageText}>{visibleSnackbarProps.message}</Text>
              </View>
              {visibleSnackbarProps.actionText && visibleSnackbarProps.actionHandler && (
                <View style={styles.actionContainer}>
                  <TouchableOpacity onPress={visibleSnackbarProps.actionHandler}>
                    <Text style={styles.actionText}>{visibleSnackbarProps.actionText}</Text>
                  </TouchableOpacity>
                </View>
              )}
            </View>
          </Animated.View>
        </FullWindowOverlay>
      )}
    </>
  );
};

Snackbar.defaultProps = {
  autoHideDuration: 4000,
  fadeInDuration: 1000,
  fadeOutDuration: 1000,
  forceFadeOutDuration: 300,
};

const styles = StyleSheet.create({
  animatedContainer: {
    flex: 1,
    position: 'absolute',
    minHeight: 48,
    paddingHorizontal: 10,
    width: '100%',
  },
  snackbar: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 5,
    backgroundColor: '#393939',
  },
  messageContainer: {
    flex: 1,
    paddingVertical: 20,
    paddingLeft: 20,
    paddingRight: 10,
  },
  actionContainer: {
    flex: 0,
    justifyContent: 'flex-end',
    alignItems: 'center',
    paddingVertical: 20,
    paddingLeft: 10,
    paddingRight: 20,
  },
  messageText: {
    color: 'white',
  },
  actionText: {
    color: '#85A3F1',
  },
});
