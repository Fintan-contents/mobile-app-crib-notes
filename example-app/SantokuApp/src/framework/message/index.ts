export * from './Message';
export * from './BundledMessageLoader';
