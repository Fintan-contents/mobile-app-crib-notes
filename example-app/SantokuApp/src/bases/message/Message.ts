/**
 * Copyright 2023 TIS Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {handleError} from 'bases/core/errors/handleError';

import {MessageKey} from './BundledMessages';
import {RuntimeError} from '../core/errors/RuntimeError';

/**
 * メッセージをロードします。
 */
interface MessagesLoader {
  load(): Promise<Record<MessageKey, string>>;
}

let cache: Record<MessageKey, string> | undefined;

/**
 * メッセージをロードします。
 * @param loader メッセージをロードするクラス
 */
async function loadMessages(loader: MessagesLoader) {
  cache = await loader.load();
}

/**
 * メッセージを取得します。
 * {@link loadMessages}を呼び出してして、メッセージをロード後に使用してください。
 * @param key メッセージのキー
 * @param options メッセージのオプション
 * @returns 指定されたメッセージキーに該当するメッセージ。指定されたメッセージキーに該当するメッセージが存在しない場合はメッセージキーを返却
 *
 * @example
 * 定義されているメッセージの例
 * ```
 * export const BundledMessages = {
 *  'validation.email': 'メールアドレスの形式が正しくありません。',
 *  'validation.required': '{0}を入力してください。',
 *  'validation.min': '{0}は{1}文字以上の値を入力してください。',
 * } as const;
 * ```
 * @example
 * メッセージを取得する例
 * ```
 * m('validation.email') // メールアドレスの形式が正しくありません。
 * m('validation.required', 'パスワード') // パスワードを入力してください。
 * m('validation.min', 'パスワード', '8') // パスワードは8文字以上の値を入力してください。
 * ```
 */
function message(key: MessageKey, ...options: string[]): string {
  if (!cache) {
    throw new Error('Messages are not loaded.');
  }
  if (cache[key] === undefined) {
    handleError(new RuntimeError(`Could not find the message. messageKey=[${key}]`, 'MessageNotFound'));
    return key;
  }
  return !options.length ? cache[key] : format(cache[key], options);
}

/**
 * メッセージをフォーマットします。
 * @param message メッセージ
 * @param options メッセージのオプション
 * @returns フォーマット後のメッセージ
 */
function format(message: string, options: string[]): string {
  return options.reduce((message, option, index) => {
    return message.replace(new RegExp(`\\{${index}\\}`, 'g'), option);
  }, message);
}

export type {MessagesLoader};
export {message as m, loadMessages};
