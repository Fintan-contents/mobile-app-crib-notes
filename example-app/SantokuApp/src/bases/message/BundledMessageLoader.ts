import {bundledMessages, BundledMessagesType} from './BundledMessages';
import {MessagesLoader} from './Message';

/**
 * アプリ内にバンドルされたメッセージをロードするクラスです。
 */
class BundledMessagesLoader implements MessagesLoader {
  load() {
    return new Promise<BundledMessagesType>(resolve => resolve(bundledMessages));
  }
}

export {BundledMessagesLoader};
