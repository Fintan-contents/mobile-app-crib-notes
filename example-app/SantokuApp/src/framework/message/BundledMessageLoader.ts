import {bundledMessages, BundledMessagesType} from '../../generated/BundledMessages';
import {MessagesLoader} from './Message';

/**
 * アプリ内にバンドルされたメッセージをロードするクラスです。
 */
class BundledMessagesLoader implements MessagesLoader {
  load() {
    return new Promise<BundledMessagesType>(resolve => resolve(bundledMessages));
  }
}

export {BundledMessagesLoader};
