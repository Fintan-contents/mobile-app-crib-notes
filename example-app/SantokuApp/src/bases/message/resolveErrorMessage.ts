import axios, {AxiosError} from 'axios';
import {m} from 'bases/message/Message';

let resolveErrorMessage = (error: unknown): {title: string; message: string} => {
  if (axios.isAxiosError(error)) {
    return resolveErrorMessageWhenAxiosError(error);
  }
  // 想定外のエラー
  return resolveErrorMessageWhenUnexpectedError();
};

function resolveErrorMessageWhenAxiosError(error: AxiosError) {
  const statusCode = error.response?.status;
  const title = m('システムエラー');
  const content = m('通信中にエラーが発生しました。');
  return {title, message: formatMessage(content, statusCode)};
}

function resolveErrorMessageWhenUnexpectedError() {
  const title = m('システムエラー');
  const content = m('しばらく経ってからもう1度お試しください。');
  return {title, message: formatMessage(content)};
}

function formatMessage(content: string, statusCode?: number) {
  if (statusCode === undefined) {
    return `${content}`;
  }
  return `${content}\n${statusCode}`;
}

const setErrorMessageResolver = (e: (error: unknown) => {title: string; message: string}) => {
  resolveErrorMessage = e;
};

export {resolveErrorMessage, setErrorMessageResolver};
