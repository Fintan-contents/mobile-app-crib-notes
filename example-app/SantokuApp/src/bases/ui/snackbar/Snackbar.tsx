import {m} from 'bases/message/Message';
import React, {useMemo, useState} from 'react';

import {
  SnackbarComponent,
  SnackbarHideProps as SnackbarComponentHideProps,
  SnackbarProps as SnackbarComponentProps,
  SnackbarShowProps as SnackbarComponentShowProps,
} from './SnackbarComponent';
import {FullWindowOverlay} from '../overlay/FullWindowOverlay';

type SnackbarShowProps = Omit<SnackbarComponentShowProps, 'message'>;
type SnackbarShowCloseButtonProps = Omit<SnackbarShowProps, 'actionText' | 'actionHandler' | 'actionTextStyle'>;
type SnackbarHideProps = Omit<SnackbarComponentHideProps, 'hide'>;

type SnackbarType = {
  /**
   * Show snackbar.
   *
   * @param message - Displayed message.
   * @param showProps - Snackbar props(SnackbarShowContextProps).
   */
  show: (message: string, showProps?: SnackbarShowProps) => void;

  /**
   * Show the snackbar with the close button.
   * The close button is placed to the right of the message.
   *
   * @param message - Displayed message.
   * @param showProps - Snackbar props(SnackbarShowCloseButtonContextProps).
   */
  showWithCloseButton: (message: string, showProps?: SnackbarShowCloseButtonProps) => void;

  /**
   * Hide snackbar.
   *
   * @param hideProps - Snackbar props(SnackbarHideContextProps).
   */
  hide: (hideProps?: SnackbarHideProps) => void;
  Component: typeof Component;
};

let Snackbar: SnackbarType = {
  show: () => {
    throw new Error('Snackbar.Component is required.');
  },
  showWithCloseButton: () => {
    throw new Error('Snackbar.Component is required.');
  },
  hide: () => {
    throw new Error('Snackbar.Component is required.');
  },
  Component,
};

function Component(props: {initialState?: SnackbarComponentShowProps}) {
  const [state, setState] = useState<SnackbarComponentProps>(
    props.initialState ?? {
      message: '',
    },
  );
  // const snackbarContext = useMemo<SnackbarContextType>(
  Snackbar = useMemo<SnackbarType>(
    () => ({
      ...Snackbar,
      show: (message: string, showProps?: SnackbarShowProps) => {
        setState({
          timestamp: Date.now(),
          ...showProps,
          message,
        });
      },
      showWithCloseButton: (message: string, showProps?: SnackbarShowCloseButtonProps) => {
        setState({
          timestamp: Date.now(),
          ...showProps,
          actionText: m('閉じる'),
          actionHandler: Snackbar.hide,
          message,
        });
      },
      hide: (hideProps?: SnackbarHideProps) => {
        setState(prevState => ({...prevState, ...hideProps, hide: true}));
      },
    }),
    [],
  );
  return (
    <>
      <FullWindowOverlay>
        <SnackbarComponent {...state} />
      </FullWindowOverlay>
    </>
  );
}

export {Snackbar};
