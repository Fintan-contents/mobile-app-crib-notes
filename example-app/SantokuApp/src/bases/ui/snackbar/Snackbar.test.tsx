import {act, render, screen} from '@testing-library/react-native';
import {BundledMessagesLoader} from 'bases/message/BundledMessageLoader';
import {loadMessages} from 'bases/message/Message';
import React, {useEffect} from 'react';
import {Text, TextStyle} from 'react-native';
import {ReactTestInstance} from 'react-test-renderer';

import {Snackbar} from './Snackbar';

// If advancing a timer changes the state of a component, the timer must be run within an act.
// However, since act is `Thenable`, ESLint will issue a warning if you do not do something like await.
// For convenience, disable the relevant rule in this file.
/* eslint-disable @typescript-eslint/no-floating-promises */

function getStyle<T>(instance: ReactTestInstance) {
  return instance.props.style as T;
}

jest.useFakeTimers();

type UseSnackbarType = 'show' | 'showWithCloseButton' | 'hide';

const ChildComponent: React.FC<{type: UseSnackbarType}> = ({type}) => {
  useEffect(() => {
    switch (type) {
      case 'show':
        Snackbar.show('テストメッセージ', {messageTextStyle: {color: 'blue'}});
        break;
      case 'showWithCloseButton':
        Snackbar.showWithCloseButton('テストメッセージ', {messageTextStyle: {color: 'red'}});
        break;
      case 'hide':
        Snackbar.hide();
    }
  }, [type]);

  return <Text testID="text">test</Text>;
};

describe('Snackbar', () => {
  it('Snackbarのshowで、Snackbarが正常に表示されることを確認', () => {
    render(
      <>
        <ChildComponent type="show" />
        <Snackbar.Component />
      </>,
    );

    expect(screen.queryByText('テストメッセージ')).not.toBeNull();
    expect(getStyle<TextStyle>(screen.getByText('テストメッセージ')).color).toBe('blue');
    expect(screen.queryByText('閉じる')).toBeNull();
    expect(screen).toMatchSnapshot();
  });

  it('SnackbarのshowWithCloseButtonで、ボタン付きSnackbarが正常に表示されることを確認', async () => {
    await loadMessages(new BundledMessagesLoader());

    render(
      <>
        <ChildComponent type="showWithCloseButton" />
        <Snackbar.Component />
      </>,
    );

    expect(screen.queryByText('テストメッセージ')).not.toBeNull();
    expect(getStyle<TextStyle>(screen.getByText('テストメッセージ')).color).toBe('red');
    expect(screen.queryByText('閉じる')).not.toBeNull();
    expect(screen).toMatchSnapshot();
  });

  it('Snackbarのhideで、Snackbarが消えることを確認', () => {
    render(
      <>
        <ChildComponent type="show" />
        <Snackbar.Component />
      </>,
    );

    screen.update(
      <>
        <ChildComponent type="hide" />
        <Snackbar.Component />
      </>,
    );

    const HIDE_FADE_OUT_DURATION = 300;
    act(() => {
      jest.advanceTimersByTime(HIDE_FADE_OUT_DURATION);
    });
    expect(screen.queryByText('テストメッセージ')).toBeNull();
    expect(screen.queryByText('閉じる')).toBeNull();
    expect(screen).toMatchSnapshot();
  });
});
