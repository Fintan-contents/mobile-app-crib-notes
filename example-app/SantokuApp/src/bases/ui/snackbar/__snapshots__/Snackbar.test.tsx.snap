// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar Snackbarのhideで、Snackbarが消えることを確認 1`] = `
Array [
  <Text
    testID="text"
  >
    test
  </Text>,
  <View
    testID="FullWindowOverlay"
  />,
]
`;

exports[`Snackbar SnackbarのshowWithCloseButtonで、ボタン付きSnackbarが正常に表示されることを確認 1`] = `
Array [
  <Text
    testID="text"
  >
    test
  </Text>,
  <View
    testID="FullWindowOverlay"
  >
    <View
      collapsable={false}
      style={
        Object {
          "bottom": 20,
          "flex": 1,
          "minHeight": 48,
          "opacity": 0,
          "paddingHorizontal": 10,
          "position": "absolute",
          "width": "100%",
        }
      }
      testID="snackbarAnimatedView"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#393939",
            "borderRadius": 5,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
        testID="snackbarStyleView"
      >
        <View
          style={
            Object {
              "flex": 1,
              "paddingLeft": 20,
              "paddingRight": 10,
              "paddingVertical": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "red",
              }
            }
          >
            テストメッセージ
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 0,
              "justifyContent": "flex-end",
              "paddingLeft": 10,
              "paddingRight": 20,
              "paddingVertical": 20,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#85A3F1",
                }
              }
            >
              閉じる
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;

exports[`Snackbar Snackbarのshowで、Snackbarが正常に表示されることを確認 1`] = `
Array [
  <Text
    testID="text"
  >
    test
  </Text>,
  <View
    testID="FullWindowOverlay"
  >
    <View
      collapsable={false}
      style={
        Object {
          "bottom": 20,
          "flex": 1,
          "minHeight": 48,
          "opacity": 0,
          "paddingHorizontal": 10,
          "position": "absolute",
          "width": "100%",
        }
      }
      testID="snackbarAnimatedView"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#393939",
            "borderRadius": 5,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
        testID="snackbarStyleView"
      >
        <View
          style={
            Object {
              "flex": 1,
              "paddingLeft": 20,
              "paddingRight": 10,
              "paddingVertical": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "blue",
              }
            }
          >
            テストメッセージ
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;
